modin/backends/pandas/query_compiler.py

```diff
@@ -626,6 +626,20 @@ def sort_index_for_equal_values(result, ascending):
 
     # END Reduction operations
 
+    def unstack(self, is_ser_out=False, is_ser_in=False, level=-1, fill_value=None):
```

Comment:
The `is_ser_out` and `is_ser_in` should not be parameters. Query compiler should always be a dataframe and should not have this interface level detail. The API layer should handle the conversion with `self._reduce_dimension` if it needs to change the dimension of the result. We try to avoid leaking these API details to the implementation.

Reply:
@devin-petersohn Query compiler has always been a dataframe in this function. Parameter `is_ser_in` I used for determination for which object this `unstack` function was called (in case of `Series` need to apply `squeeze()` for `df` in `map_func`). Parameter `is_ser_out` I used for excluding of recomputing `new_columns` in function `apply_full_axis` (in case of `Series` `new_columns` was set to ["__reduced__"]).

`self._reduce_dimension` doesn't solve my problem because this function calls constructor of `Series` internally and result-object contains `name = self._query_compiler.columns[0]` which should be `None` in case of calling `unstack` function in my case. Therefore, I added parameter `new_columns` to `unstack` in `query_compiler` and set its into `["__reduced__"]` when result should be a `Series` ([line](https://github.com/modin-project/modin/pull/1649/files#diff-73ebd3f1e2f845adf0a4281cb08bf6bcR1561)).